dissect/tests/test_fat32.py

```diff
@@ -0,0 +1,434 @@
+
+            with self.assertRaises(fat32.FileDoesNotExistException):
+                fs.delDirectory("/test-longlong")
+
+            # this value extracted from an entry created by the Windows FAT32 driver
+            self.assertEqual(fat32.DIRECTORY_DATA.compute83Hash("TEST-L~1   "), 0x7)
+
+            fs.addDirectory("/test-longlonglonglong")
+            self.assertEqual(list(fs.listDirectories()), ["/test-longlonglonglong"])
+
+            with self.assertRaises(fat32.FileExistsException):
+                fs.addDirectory("/test-longlonglonglong")
+
+            fs.addDirectory("/test-longlonglonglong/hahahahahahahahahaha")
+            self.assertEqual(list(fs.listDirectories()), ["/test-longlonglonglong", "/test-longlonglonglong/hahahahahahahahahaha"])
+
+            with self.assertRaises(fat32.DirectoryNotEmptyException):
+                fs.delDirectory("/test-longlonglonglong")
+
+            fs.delDirectory("/test-longlonglonglong/hahahahahahahahahaha")
+            self.assertEqual(list(fs.listDirectories()), ["/test-longlonglonglong"])
+
+            fs.delDirectory("/test-longlonglonglong")
+            self.assertEqual(list(fs.listDirectories()), [])
+
+    def test_files(self):
+        with test_logical_fs() as fs:
+            self.assertEqual(list(fs.listFiles()), [])
+            self.assertEqual(list(fs.listDirectories()), [])
+            with self.assertRaises(fat32.FileDoesNotExistException):
+                fs.readFile("/dne.txt")
+
+            with self.assertRaises(fat32.FileDoesNotExistException):
+                fs.delFile("/test-longlong.txt")
+
+            fs.addFile("/test-longlong.txt", b"AA")
+            self.assertEqual(list(fs.listFiles()), ["/test-longlong.txt"])
+            self.assertEqual(fs.readFile("/test-longlong.txt"), b"AA")
+
+            with self.assertRaises(fat32.FileExistsException):
+                fs.addFile("/test-longlong.txt", b"AA")
+
+            fs.delFile("/test-longlong.txt")
+            self.assertEqual(list(fs.listFiles()), [])
+
+            fs.addDirectory("/test-longlonglonglong")
+            self.assertEqual(list(fs.listDirectories()), ["/test-longlonglonglong"])
+
+            fs.addFile("/test-longlonglonglong/test-longlong.txt", b"AA")
+            self.assertEqual(list(fs.listFiles()), ["/test-longlonglonglong/test-longlong.txt"])
+            self.assertEqual(fs.readFile("/test-longlonglonglong/test-longlong.txt"), b"AA")
+
+            with self.assertRaises(fat32.DirectoryNotEmptyException):
+                fs.delDirectory("/test-longlonglonglong")
+            self.assertEqual(list(fs.listFiles()), ["/test-longlonglonglong/test-longlong.txt"])
+
+            fs.delFile("/test-longlonglonglong/test-longlong.txt")
+            self.assertEqual(list(fs.listFiles()), [])
+            fs.delDirectory("/test-longlonglonglong")
+
+
+def test():
+    logging.basicConfig(level=logging.DEBUG)
+    try:
+        unittest.main()
+    except SystemExit:
+        pass
+
```

Comment:
maybe functionality like this ( setting logging to debug and running unittest main() ) should be a function in something like dissect/tests/common.py ?  I like the idea of keeping the debug output clean when using python -m unittest discover, but having an option to python -m <testmod> to cause debug output running.  Clever.